NSS-Day-Cohort-68/rare-api-rare-team-4b
# Added GET functionality for retrieving all posts

This pull request adds the functionality to retrieve all posts via a GET request to `/posts`.
- Added GET route handler for `/posts` endpoint in `server.py`.
- Implemented logic to fetch all posts from the database.
- Tested the functionality locally to ensure it works as expected.

Testing:
1. Run `server.py`.
2. Make a GET request in Postman at `/posts` to verify that all posts are retrieved successfully.

Ticket #36
